adafruit_slideshow.py

```diff
@@ -0,0 +1,209 @@
+        return self._order
+
+    @order.setter
+    def order(self, order):
+        self._order = order
+
+    @property
+    def loop(self):
+        """Specifies whether to loop through the images continuously or play through the list once.
+        ``True`` will continue to loop, ``False`` will play only once. Default is `True`."""
+        return self._loop
+
+    @loop.setter
+    def loop(self, loop):
+        self._loop = loop
+
+    @property
+    def dwell(self):
+        """The number of seconds each image displays, in seconds. Default is 3."""
+        return self._dwell
+
+    @dwell.setter
+    def dwell(self, dwell):
+        self._dwell = dwell
+
+    def update(self):
+        """Updates the slideshow to the next image."""
+        now = time.monotonic()
+        if self._current_state == self.FADE_IN:
+            steps = 100
+            for b in range(steps):
+                self._backlight.duty_cycle = b * SlideShow.MAX_BRIGHTNESS // steps
+                time.sleep(0.01)
+            self._current_state = self.SHOW_IMG
+            self._img_start = time.monotonic()
+
+        if self._current_state == self.SHOW_IMG:
+            if now - self._img_start > self._dwell:
+                self._current_state = self.FADE_OUT
+
+        if self._current_state == self.FADE_OUT:
+            steps = 100
+            for b in range(steps, -1, -1):
+                self._backlight.duty_cycle = b * SlideShow.MAX_BRIGHTNESS // steps
+                time.sleep(0.01)
+            self._group.pop()
+            self._current_state = self.LOAD_IMG
+
+        if self._current_state == self.LOAD_IMG:
+            try:
+                imagename = next(self._images)
+            except StopIteration:
+                return False
+            try:
+                self._show_bmp(imagename)
+                self._current_state = self.FADE_IN
+            except ValueError as error:
+                print("Incompatible image:", imagename, str(error))
+
+        return True
+
+    def _show_bmp(self, imagename):
+        """Opens and loads the image onto the display."""
+        with open(imagename, "rb") as image:
+            odb = displayio.OnDiskBitmap(image)
+            face = displayio.Sprite(odb, pixel_shader=displayio.ColorConverter(), position=(0, 0))
+            self._group.append(face)
+            board.DISPLAY.wait_for_frame()
+
+    def _get_next_image(self):
+        """Cycles through the list of images."""
+        while True:
+            for image in self._file_list:
+                yield image
+            if not self._loop:
+                return
+
+    def _load_images(self):
+        """Loads the list of images to be displayed in alphabetical or random order."""
+        self._file_list = self._get_filenames()
+        if self._order == SlideShow.RANDOM:
+            self._file_list = sorted(self._file_list, key=lambda x: random.random())
+            self._images = self._get_next_image()
```

Comment:
Unindent. Shouldn't be in the `if`. Looks like that was my bad. I only tested random. :(

Reply:
I was testing it with alphabetical and it seemed to be working so I didn't think anything of it. On closer investigation it was only mostly working. Fixed.